addons/mail/models/mail_tracking_value.py

```diff
@@ -61,7 +61,7 @@ def create_tracking_values(self, initial_value, new_value, col_name, col_info):
                 'old_value_integer': initial_value and initial_value.id or 0,
                 'new_value_integer': new_value and new_value.id or 0,
                 'old_value_char': initial_value and initial_value.name_get()[0][1] or '',
-                'new_value_char': new_value and new_value.name_get()[0][1] or ''
+                'new_value_char': new_value and new_value.sudo().name_get()[0][1] or ''
```

Comment:
Not sure about this one since the `sudo` could be applied on any object, in a public method.